### PR TITLE
[MRG] Fix dgap freq

### PR DESCRIPTION
**Pull request by mathurinm**
@joewalter

### CODE REVIEW

File: mne/inverse_sparse/mxne_optim.py [hunk 1]
```diff
@@ -297,6 +297,7 @@ def _mixed_norm_solver_cd(M, G, alpha, lipschitz_constant, maxit=10000,
                           dgap_freq=10):
     """Solve L21 inverse problem with coordinate descent."""
     from sklearn.linear_model.coordinate_descent import MultiTaskLasso
+    warn('dgap_freq parameter is ignored when scikit learn is used as solver.')
```
Comment (agramfort) [inline mne/inverse_sparse/mxne_optim.py]: so now I get a warning all the time???
it's bad. Then it should be set to 'auto' by default and only report a warning when changed by user and algo is using sklearn

Comment (agramfort) [reply mne/inverse_sparse/mxne_optim.py]: actually sklearn uses also dgap_freq of 10 so I would not bother actually

I would keep the silent ignore

Comment (mathurinm) [reply mne/inverse_sparse/mxne_optim.py]: sklearn uses a dgap_freq ? I thought the dgap computation was triggered only if the update was small, not perdiodically*
I am OK with silent warning

Comment (agramfort) [reply mne/inverse_sparse/mxne_optim.py]: indeed my bad. I would just ignore this silently. sklearn code will be super fast anyway. So just add a note to the docstring.

**Pull request merged**